libvmi/libvmi_extra.h

```diff
@@ -56,6 +56,18 @@ GSList* vmi_get_va_pages(
 #ifdef LIBVMI_EXTRA_JSON
 #include <json-c/json.h>
 
+/**
+ * Initialize kernel's JSON profile from particular config type.
+ * @param[in] vmi Instance
+ * @param[in] config_mode The type of OS configuration that is provided.
+ * @param[in] config Configuration is passed directly to LibVMI (ie. in a string
+ *                   or in a GHashTable) or NULL of global config file is used.
+ */
+os_t vmi_init_profile(
+    vmi_instance_t vmi,
+    vmi_config_t config_mode,
+    void *config) NOEXCEPT;
```

Comment:
I don't understand this function. You wanted to initialize the JSON interface, but then why would this function take a string or a ghashtable as input instead of a `const char* path` to the json file?

Reply:
I guess I would be OK with this since it matches the other init functions input params. But I think you need to add more information what this function actually enables and what won't work the way a user would expect. For example, while enabling the json parser routines will get you offsets and structure sizes, things like `vmi_get_symbol_addr_from_json` won't work the same way as when the os part of LibVMI is initialized since we don't yet know the KASLR offset.

So I think there needs to be better description before we can add this to the API.

Reply:
Actually never mind that example, for some reason I thought that function adjusts the values to KASLR but that's the other LibVMI symbol translation function that does that. I guess I should get some sleep :P

Reply:
I've added some more explanations.